pds_doi_core/util/doi_validator.py

```diff
@@ -46,6 +48,25 @@ def get_database_name(self):
     def __lidvid(columns, row):
         return f"{row[columns.index('lid')]}::{row[columns.index('vid')]}"
 
+    def _check_field_site_url(self, doi: Doi):
+        """ If the site_url field exist in doi object, check to see if it is online.  If the site is not online, an exception will be thrown.
+        """
+
+        logger.debug(f"doi {doi}")
+        logger.info(f"doi.site_url {doi.site_url}")
+
+        if doi.site_url:
+            try:
+                request = requests.get(doi.site_url,timeout=5)
+                logger.debug(f"site_url {doi.site_url} indeed exist")
+            except (requests.exceptions.ConnectionError, Exception) as e:
```

Comment:
You should also check for http errors.
For example url https://pds.jpl.nasa.gov/ds-view/pds/viewBudle.jsp?identifier=urn%3Anasa%3Apds%3Ainsight_camera&amp;version=1.0 send http error 404 (not found).

The error code is available in response.status_code. if it is >=400 an exception should be raised.

I am also checking with Jordan if we should capture for such results as errors: https://pds.jpl.nasa.gov/ds-view/pds/viewBundle.jsp?identifier=urn%3Anasa%3Apds%3Ainsight_camera&amp;version=1.0

Theoretically we should but that will take more time to implement, so we'll see.

Reply:
Setting the site_url as the one you'd mentioned with this code snippet seems to work.  The code caught the 404 and raise the exception:

        if doi.site_url:
            try:
                response = requests.get(doi.site_url,timeout=5)
                status_code = response.status_code
                logger.debug(f"from_request:status_code,site_url {status_code,doi.site_url}")
                # Handle cases when a connection can be made to the server but the status is greater than or equal to 400.
                if status_code >= 400:
                    # Need to check its an 404, 503, 500, 403 etc.
                    raise requests.HTTPError(f"status_code,site_url {status_code,doi.site_url}")
                else:
                    logger.debug(f"site_url {doi.site_url} indeed exist")
            except (requests.exceptions.ConnectionError, Exception) as e:
                error_message = f"site_url {doi.site_url} not exist"
                # Although not being able to connect is an error, the message printed is a warning.
                logger.warning(error_message)
                raise SiteURNotExistException(error_message)

This is the response from the code as expected.

    raise WarningDOIException(str(e))
pds_doi_core.input.exceptions.WarningDOIException: SiteURNotExistException:site_url https://pds.jpl.nasa.gov/ds-view/pds/viewBudle.jsp?identifier=urn%3Anasa%3Apds%3Ainsight_camera&amp;version=1.0 not exist, SiteURNotExistException:site_url https://pds.jpl.nasa.gov/ds-view/pds/viewBudle.jsp?identifier=urn%3Anasa%3Apds%3Ainsight_camera&amp;version=1.0 not exist, SiteURNotExistException:site_url https://pds.jpl.nasa.gov/ds-view/pds/viewBudle.jsp?identifier=urn%3Anasa%3Apds%3Ainsight_camera&amp;version=1.0 not exist

Reply:
good, thanks